MekHQ/src/mekhq/campaign/parts/Part.java

```diff
@@ -1058,8 +1072,16 @@ public boolean canNeverScrap() {
 	}
 
 	public String scrap() {
+		String msg = "";
+		
+		if (null == getUnit()) {
+			msg = getName() + " scrapped.";
+		} else {
+			msg = getName() + " on " + unit.getName() + " scrapped.";
+		}
+		
 		remove(false);
-		return getName() + " scrapped.";
+		return msg;
```

Comment:
Actually, that won't work. The call to remove() is done last, after the msg is built, because the remove() call sets the unit to null, thus you'll never have the unit's name in the message.